manager/integration/tests/test_node.py

```diff
@@ -0,0 +1,359 @@
+    assert len(volumes) == 0
+    # clean all disks
+    common.cleanup_disk()
+
+
+def generate_update_disk(disks):
+    update_disk = []
+    for key, disk in disks.iteritems():
+        update_disk.append(disk)
+    return update_disk
+
+
+def test_node(clients):  # NOQA
+    # get a random client
+    for host_id, client in clients.iteritems():
+        break
+
+    # test node update
+    nodes = client.list_node()
+    assert len(nodes) > 0
+    lht_hostId = get_self_host_id()
+    node = client.by_id_node(lht_hostId)
+    node = client.update(node, allowScheduling=False)
+    assert not node["allowScheduling"]
+    node = client.by_id_node(lht_hostId)
+    assert not node["allowScheduling"]
+
+    node = client.update(node, allowScheduling=True)
+    assert node["allowScheduling"]
+    node = client.by_id_node(lht_hostId)
+    assert node["allowScheduling"]
+
+    disks = node["disks"]
+
+    # test add and delete exception
+    with pytest.raises(Exception):
+        # add same disk by different mount path
+        disk = {"path": "/var/lib", "allowScheduling": True,
+                "storageMaximum": 5*Gi, "storageReserved": 2*Gi}
+        update_disk = generate_update_disk(disks)
+        update_disk.append(disk)
+        node = node.diskUpdate(disks=update_disk)
+        # delete exception
+        node = node.diskUpdate(disks=[])
+
+    # create multiple disks for node
+    node = client.by_id_node(lht_hostId)
+    disks = node["disks"]
+    disk_path1 = create_disk(client, 'vol-disk-1',
+                             SIZE, lht_hostId)
+    disk1 = {"path": disk_path1, "allowScheduling": True,
+             "storageMaximum": 5*Gi, "storageReserved": 2*Gi}
+    disk_path2 = create_disk(client, 'vol-disk-2',
+                             SIZE, lht_hostId)
+    disk2 = {"path": disk_path2, "allowScheduling": True}
+
+    update_disk = generate_update_disk(disks)
+    # add new disk for node
+    update_disk.append(disk1)
+    update_disk.append(disk2)
+
+    # save disks to node
+    node = node.diskUpdate(disks=update_disk)
+    assert len(node["disks"]) == len(update_disk)
+
+    node = client.by_id_node(lht_hostId)
+    assert len(node["disks"]) == len(update_disk)
+
+    # update disk
+    disks = node["disks"]
+    update_disk = generate_update_disk(disks)
+    for disk in update_disk:
+        # keep default disk for other tests
+        if disk["path"] == disk_path1 or disk["path"] == disk_path2:
+            disk["allowScheduling"] = False
+            disk["storageReserved"] = Gi
+    node.diskUpdate(disks=update_disk)
+    node = client.by_id_node(lht_hostId)
+    disks = node["disks"]
+    for key, disk in disks.iteritems():
+        if disk["path"] == disk_path1 or disk["path"] == disk_path2:
+            assert not disk["allowScheduling"]
+            assert disk["storageReserved"] == Gi
+            assert disk["storageScheduled"] == 0
+
+    # delete disk
+    update_disk = generate_update_disk(disks)
+    delete_disk = []
```

Comment:
Um, don't call it `delete_disk`. It sounds like the disk going to be deleted. It's more like the `remain_disk`.